### PR TITLE
ui: metrics load

### LINKED ISSUE #44836: ui: metrics load slowly when sharing URLs with large time spans
**Describe the problem**

When sharing URLs to metrics graphs with large (>day) windows, the rescaling is not preserved from the original page and results in a very large number of queries running, bringing the browser to a halt.

**To Reproduce**

1. Set up CockroachDB cluster ...
2. Run load for a day
3. Open the default metrics view to a 2d interval
4. Copy/paste the URL to a new browser window
5. Observe long wait time until graph loads

Kapture 2020-02-06 at 16 33 19

**Expected behavior**
The graphs should load as quickly as they do on the original browser tab where the user simply clicks "2d" in the dropdown

Kapture 2020-02-06 at 16 29 50

**Environment:**
 - CockroachDB version 20.1 Alpha
 - Admin UI

### PR DESCRIPTION
**Pull request by vladlos**
Optimised metrics data loading when sharing URLs to metrics graphs with large timescale

Resolves: #44836

Release note(ui): none

### CODE REVIEW

File: pkg/ui/src/views/devtools/containers/raftMessages/index.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ export class RaftMessages extends React.Component<RaftMessagesProps> {
     this.setClusterPath(selected.value);
   }
 
-  componentWillMount() {
+  componentDidMount() {
```
Comment (dhartunian) [inline pkg/ui/src/views/devtools/containers/raftMessages/index.tsx]: I know these are good changes to make but if you want to address React upgrade warnings, put them in a separate PR.

**Review by dhartunian — CHANGES_REQUESTED** [review]

File: pkg/ui/src/views/cluster/containers/timescale/index.tsx [hunk 2]
```diff
@@ -207,14 +206,22 @@ class TimeScaleDropdown extends React.Component<TimeScaleDropdownProps, {}> {
...
+  findClosestTimeScale = (seconds: number) => {
+    const data: timewindow.TimeScale[] = [];
+    Object.keys(availableTimeScales).forEach((val) => data.push(availableTimeScales[val]));
+    data.sort( (a, b) => (Math.abs(seconds - a.windowSize.asSeconds()) - Math.abs(seconds - b.windowSize.asSeconds())) );
+    return data[0];
+  }
 
-    selected.key = "Custom";
-    this.props.setTimeScale(selected);
+  setDatesByQueryParams = (dates?: timewindow.TimeWindow) => {
```
[the remark this answers is not in the document]
Comment (elkmaster) [reply pkg/ui/src/views/cluster/containers/timescale/index.tsx]: Main idea was to leave url human readable or even user generated. Model of timewindow consists from 3 fields, including labeling. Pass the whole object will be messy. And passing only part of that will require very similar calculations.

**Comment by dhartunian** [comment]
@elkmaster test doesn't currently pass for me, can you take a look?

**Review by dhartunian — APPROVED** [review]
